core/src/util/builders.cpp

```diff
@@ -137,6 +137,12 @@ void Builders::buildPolygonExtrusion(const Polygon& _polygon, float _minHeight,
             normalVector = glm::cross(upVector, b - a);
             normalVector = glm::normalize(normalVector);
 
+            if (std::isnan(normalVector.x)
```

Comment:
I believe that only the MVT decoder removes duplicate points. Checking in the building process is probably more robust in general, e.g. client-provided data could have degenerate geometry.